agentMET4FOF/metrological_agents.py

```diff
@@ -356,4 +353,833 @@ def agent_loop(self):
         """
         if self.current_state == "Running":
             self.set_output_data(channel="default", data=self._stream.next_sample())
-            super().agent_loop()
\ No newline at end of file
+            super().agent_loop()
+
+
+class RedundancyAgent(MetrologicalAgent):
+    """
+    This is the main Redundancy Agent class. Main calculation types are :py:func:`lcs` and :py:func:`lcss`, as defined
+    in the module :mod:`redundancy1`.
+    """
+
+    def init_parameters(
+        self,
+        input_data_maxlen: int = 25,
+        output_data_maxlen: int = 25,
+        sensor_key_list: list = None,
+        n_pr: int = 1,
+        problim: float = .9,
+        calc_type: str = 'lcs'
+    ):
+        """
+        Initialize the redundancy agent as an instance of the :py:mod:`MetrologicalAgent` class.
+
+
+        Parameters
+        ----------
+        n_pr : int, optional
+            size of the batch of data that is handled at a time by the Redundancy Agent. Defaults to 1
+        problim : float, optional
+            limit probability used for consistency evaluation. Defaults to .9
+        calc_type : str, optional
+            calculation type: 'lcs' or 'lcss'. Defaults to 'lcs'
+        sensor_key_list : list of str
+            list containing the names of the sensors that should feed data to the Redundancy Agent. Defaults to None
+
+        Parent class parameters
+        ----------
+        input_data_maxlen: int
+
+        output_data_maxlen: int
+
+        """
+
+        if sensor_key_list is None:
+            sensor_key_list = []
+        super().init_parameters(input_data_maxlen=25, output_data_maxlen=25)
+        self.metadata = MetaData(
```

Comment:
I would suggest to add type hints and declare those instance variables at the top of the class with for instance:

```python
metadata: MetaData
calc_type: str
[...] 
```